lib/widgets/drive/index.coffee

```diff
@@ -0,0 +1,119 @@
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+###
+
+_ = require('lodash')
+async = require('async')
+Promise = require('bluebird')
+drivelist = Promise.promisifyAll(require('drivelist'))
+form = require('resin-cli-form')
+InquirerList = require('inquirer/lib/prompts/list')
+UI = require('inquirer/lib/ui/baseUI')
+DriveScanner = require('./drive-scanner')
+
+_format = '<%= device %> (<%= size %>) - <%= description %>'
+
+getDrives = ->
+	drivelist.listAsync().then (drives) ->
+		Promise.fromNode (callback) ->
+			async.reject drives, drivelist.isSystem, (results) ->
+				return callback(null, results)
+
+cleanupList = (previous,currrent) ->
+	removedDrive =
+		name: _.template(_format)(currrent)
+		value: currrent.device
+
+	return _.reject previous, (drive) ->
+
+		# The reason we use _.pick and pass a custom object in _.isEqual is due to
+		# the fact the initial objects (previous, current) originate from different
+		# prototypes i.e. previous from Inquirer's Choice class and
+		# current/removedDrive from the results provided by drivelist. This makes
+		# a direct deep comparison between them impossible due the different
+		# properties that they contain.
+		return _.isEqual(_.pick(drive, 'name', 'value'), removedDrive)
+
+###*
+# @summary Prompt the user to select a drive device
+# @name visuals.drive
+# @function
+# @public
+# @memberof visuals
+#
+# @description
+# The dropdown detects and autorefreshes itself when the drive list changes.
+#
+# @param {String} [message='Select a drive'] - message
+# @returns {Promise<String>} device path
+#
+# @example
+# visuals.drive('Please select a drive').then (drive) ->
+# 	console.log(drive)
+###
+module.exports = (message = 'Select a drive') ->
+	getDrives().then (drives) ->
+		options =
+			message: message
+
+			# Even though the name property doesn't have an actual use in the code
+			# Inquirer forces us to declare it. Skipping an explicit choices
+			# declaration would cause an Error during the InquirerList instantiation.
+			name: 'drives'
+			choices: _.map drives, (drive) ->
+				return {
+					name: _.template(_format)(drive)
```

Comment:
Hint: You can create a function that takes a `drive` as an argument and returns the object. This way you can shorten `choices` to:

```coffee
choices: _.map(drives, getDriveChoice)
```
```